query/src/main/java/com/connexta/replication/query/QueryService.java

```diff
@@ -0,0 +1,223 @@
+    synchronized (this) {
+      this.site = site;
+      try {
+        this.adapter = queryServiceTools.getAdapterFor(site);
+      } catch (AdapterException e) {
+        LOGGER.info(
+            "Failed to create an adapter for {} stopping the query service for now."
+                + site.getUrl());
+        stop();
+      }
+    }
+  }
+
+  /** Starts this query service. */
+  public void start() {
+    if (!isRunning()) {
+      future =
+          executor.scheduleAtFixedRate(
+              this::query, 0, queryServiceTools.getPeriod(), TimeUnit.SECONDS);
+    }
+  }
+
+  /** Stops this query service. */
+  public void stop() {
+    executor.shutdownNow();
+  }
+
+  @VisibleForTesting
+  void query() {
+    if (!adapter.isAvailable()) {
+      LOGGER.debug("System at " + site.getUrl() + " is currently unavailable");
+      return;
+    }
+
+    Set<Filter> filters = queryServiceTools.activeFiltersFor(site).collect(Collectors.toSet());
+    for (Filter filter : filters) {
+      FilterIndex filterIndex = queryServiceTools.getOrCreateFilterIndex(filter);
+
+      // holds the modified time of each task put in the queue for this filter
+      // this will be used to grab the latest modified time after
+      Set<Instant> modifiedTimes = new HashSet<>();
+
+      try {
+        getNewMetadataForFilter(filter, filterIndex)
+            .map(metadatum -> metadataToTask(metadatum, filter.getPriority()))
+            .forEach(task -> queueTaskAndPutLastModifiedInSet(task, modifiedTimes));
+      } catch (AdapterException e) {
+        LOGGER.debug(
+            "Couldn't query site {}. Stopping queries for this site until next polling attempt.",
+            site.getName(),
+            e);
+        return;
+      } catch (QueueException qe) {
+        LOGGER.debug(
+            "Failed to put a task in queue for site {}. Stopping queuing of tasks for this site. Will resume at next polling attempt.",
+            site.getName(),
+            qe);
+        return;
+      }
+      saveLastModifiedInFilterIndex(filterIndex, modifiedTimes);
+    }
+  }
+
+  private Stream<Metadata> getNewMetadataForFilter(Filter filter, FilterIndex filterIndex)
+      throws AdapterException {
+    // get last modified date
+    final Date indexDate = filterIndex.getModifiedSince().map(Date::from).orElse(null);
+    if (indexDate == null) {
+      LOGGER.trace("No index for filter {} found.", filter.getName());
+    }
+
+    // not implementing failure retries here
+    QueryRequest queryRequest =
+        new QueryRequestImpl(
+            filter.getFilter(),
+            Collections.emptyList(), // ignoring excluded nodes (origins tag)
+            Collections.emptyList(), // ignoring failed items
+            indexDate);
+
+    Iterable<Metadata> changeSet = adapter.query(queryRequest).getMetadata();
+    return StreamSupport.stream(changeSet.spliterator(), false);
+  }
+
+  private TaskInfo metadataToTask(Metadata metadata, byte filterPriority) {
```

Comment:
❗️ One thing that will need to be enhanced later here is the fact that for regional DDF sites, we don't have to handle deleted pieces of intel since we are only harvesting from those whereas for other combinations, we do have to handle deletes and this means we will need to figure out a way to get that info and propagate delete operations.

For now, we should ensure to filter out any sites that are not regional DDF from the list of sites we start query services for inside the manager.